hal/src/boron/network/sara_ncp_client.cpp

```diff
@@ -155,46 +155,68 @@ int SaraNcpClient::initParser(Stream* stream) {
             .commandTerminator(AtCommandTerminator::CRLF);
     parser_.destroy();
     CHECK(parser_.init(std::move(parserConf)));
+    // +CREG: <stat>[,<lac>,<ci>[,<AcTStatus>]]
     CHECK(parser_.addUrcHandler("+CREG", [](AtResponseReader* reader, const char* prefix, void* data) -> int {
         const auto self = (SaraNcpClient*)data;
-        int val[2];
-        int r = CHECK_PARSER_URC(reader->scanf("+CREG: %d,%d", &val[0], &val[1]));
+        int val[4];
+        // Parse as direct response (ignoring mode)
+        int r = CHECK_PARSER_URC(reader->scanf("+CREG: %*d,%d,\"%x\",\"%x\",%d", &val[0], &val[1], &val[2], &val[3]));
```

Comment:
Looking at this more, <stat> would be val[1] for this direct response with verbose output <n>=2, whereas <stat> would be val[0] for the URC.

Reply:
I'm specifically ignoring _<mode>_ `%*d`, so `val[0]` must always be _<stat>_.

Reply:
I have also reduced the tolerance for matching, so this should only parse the appropriate values.

Reply:
@technobly Would you review the new logic and see if it addresses your concerns?